src/reader/reader.h

```diff
@@ -195,9 +197,8 @@ class InmemReader : public Reader {
   // Free the memory of data matrix.
   virtual void Clear() {
     data_buf_.Reset();
-    data_samples_.Reset();
```

Comment:
`data_samples_.Reset()` will clear some memory, why remove this line of code? Thanks.

Reply:
As I was debugging and walking through the code, I discovered that the `Row` variable in both `data_buf` and `data_sample` are storing pointers that point to the same memory.

Besides, even though the `Reset` method is called on one DMatrix, the `Row` variable in another DMatrix will not be reset. This causes the `Row` variable to keep storing SparseRow pointers pointing to the memory that was already freed.

Calling the `Reset` method again on `data_sample` will cause a segfault. So, I removed `data_sample.Reset();` because this line is unnecessary.

Reply:
Is it because of ``Clear`` added to destructor of ``InmemReader``. I found that we use default destructor for ``InmemReader`` before.